common/src/main/java/org/vivecraft/mixin/client/multiplayer/ClientPacketListenerVRMixin.java

```diff
@@ -48,9 +48,12 @@ public class ClientPacketListenerVRMixin {
 
     @Inject(at = @At("TAIL"), method = "<init>")
     public void init(Minecraft minecraft, Screen screen, Connection connection, GameProfile gameProfile, ClientTelemetryManager clientTelemetryManager, CallbackInfo ci) {
-        ClientDataHolder.getInstance().vrSettings.overrides.resetAll();
-        ClientNetworkHelper.resetServerSettings();
-        ClientNetworkHelper.displayedChatMessage = false;
+        if (ClientNetworkHelper.needsReset) {
+            ClientDataHolder.getInstance().vrSettings.overrides.resetAll();
+            ClientNetworkHelper.resetServerSettings();
+            ClientNetworkHelper.displayedChatMessage = false;
+            ClientNetworkHelper.needsReset = false;
+        }
```

Comment:
Why is the reset needed

Reply:
the reset was already there, hadn't checked
but those resets are there, so we know to re ask the server for the settings

Reply:
now that I think of it that extra check probably should also be added to the nonvr version

Reply:
It's a mixin into the init, so it will only be called once right? is the needsReset needed? What did this solve?

Reply:
figura has it's own client Packet listener, so it would be called twice